src/libraries/System.Private.CoreLib/src/System/IO/File.cs

```diff
@@ -343,7 +343,7 @@ public static IAsyncEnumerable<string> ReadLinesAsync(string path, Encoding enco
         {
             Validate(path, encoding);
 
-            var sr = new DisposingFlagStreamReader(path, encoding); // Move first streamReader allocation here so to throw related file exception upfront, which will cause known leaking if user never actually foreach's over the enumerable
+            StreamReader sr = AsyncStreamReader(path, encoding); // Move first streamReader allocation here so to throw related file exception upfront, which will cause known leaking if user never actually foreach's over the enumerable
```

Comment:
What if they used linq query/method style with the function instead of foreach? Will it leak in that case?